### PR TITLE
Implement feature #2:Add expense to CSV

### LINKED ISSUE #2: Issue 2 — Append expense to CSV (`ExpenseStore.append`)
**Goal:** Add a new expense row to `data/expenses.csv`.

**Details:**
- If file doesn’t exist, create it and write the header first.
- Then append the new row.

**Acceptance Criteria:**
- First append creates file + header + row.
- Second append adds one more row.
- `load` reads appended rows correctly.

### PR DESCRIPTION
**Pull request by bbirribbarribbo**
This PR solves #2

- Implemented the CSV file writing logic in the ExpenseStore append method.
- Implemented the expenseToCSV method to convert an Expense object into a CSV string.
- Added the ExpenseStoreTest file

How to test: add an expense with an empty participants list → check the CSV file → verify the participants column for that new line is empty.

happy_path:
1. Initial write to a new file
2. Appending to an existing file

edge_case:
empty participant list

Closes #2

### CODE REVIEW

**Comment by hayat711** [comment]
**Group6** will be reviewing this PR [code implemented by group7 for feature #2 ]

File: src/main/java/itm/oss/splitter/ExpenseStore.java [hunk 1]
```diff
@@ -1,25 +1,71 @@
...
+    // Create parent directories if they don't exist
+    File parentDir = file.getParentFile();
+    if (parentDir != null && !parentDir.exists()) {
+      parentDir.mkdirs();
+    }
+    
+    boolean fileExists = file.exists() && file.length() > 0;
+    
+    try (FileWriter fw = new FileWriter(file, true); // append mode
+         PrintWriter pw = new PrintWriter(fw)) {
```
Comment (gunuzello) [inline src/main/java/itm/oss/splitter/ExpenseStore.java]: Maybe we can make the file use UTF-8 when writing.  
It will help avoid weird characters if someone runs this on another computer.

File: src/main/java/itm/oss/splitter/ExpenseStore.java [hunk 2]
```diff
@@ -1,25 +1,71 @@
...
+        
+        // CSV format
+        return String.join(",",
+                e.getDate(),
+                e.getPayer(),
+                String.valueOf(e.getAmount()),
+                e.getCurrency(),
+                participantsStr,
+                e.getCategory(),
+                e.getNotes()
```
Comment (gunuzello) [inline src/main/java/itm/oss/splitter/ExpenseStore.java]: It looks clean, but maybe we can add a small helper method like csv()  
to make sure each field is safe for CSV files.

File: src/main/java/itm/oss/splitter/ExpenseStore.java [hunk 3]
```diff
@@ -1,25 +1,71 @@
...
+import java.io.FileWriter;
 import java.io.IOException;
+import java.io.PrintWriter;
 import java.util.ArrayList;
 
 public class ExpenseStore {
 
+  public static final String HEADER = "date,payer,amount,currency,participants,category,notes";
+
   public ArrayList<Expense> load(String path) throws IOException {
```
Comment (gunuzello) [inline src/main/java/itm/oss/splitter/ExpenseStore.java]: Maybe we can return List<Expense> instead of ArrayList<Expense>  
so it can work with other list types later.

**Review by gunuzello — COMMENTED** [review]
I focused on readability and file handling. 
I suggested using UTF-8 for file writing and adding CSV safety checks. 
Also recommended using List instead of ArrayList for flexibility.

File: src/test/java/itm/oss/splitter/ExpenseStoreTest.java [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+import org.junit.jupiter.api.Test;
+
+class ExpenseStoreTest {
+
+    private ExpenseStore store;
+    private Expense sampleExpense1;
+    private Expense sampleExpense2;
+
+    // 테스트용 파일 이름을 정해둡니다. (프로젝트 폴더 최상위에 생깁니다)
+    private static final String TEST_FILE_PATH = "my_test_expenses.csv";
```
Comment (ohkyounghun) [inline src/test/java/itm/oss/splitter/ExpenseStoreTest.java:34]: Using a fixed file in the repo root can leave artifacts and cause flaky tests across environments.
Consider JUnit 5 @TempDir to get an isolated temporary directory per test and avoid manual delete logic.

File: src/test/java/itm/oss/splitter/ExpenseStoreTest.java [hunk 5]
```diff
@@ -0,0 +1,170 @@
...
+        // 첫 번째 데이터를 미리 추가해둠 (헤더 + 데이터1)
+        store.append(TEST_FILE_PATH, sampleExpense1);
+
+        // 2. When (실행)
+        // 두 번째 데이터를 추가
+        store.append(TEST_FILE_PATH, sampleExpense2);
+
+        // 3. Then (검증)
+        File testFile = new File(TEST_FILE_PATH);
+        try (Scanner scanner = new Scanner(testFile)) {
```
Comment (ohkyounghun) [inline src/test/java/itm/oss/splitter/ExpenseStoreTest.java]: It might be safer to specify the charset explicitly, e.g., new Scanner(testFile, StandardCharsets.UTF_8),
so the tests don’t depend on the platform default encoding (prevents failures on different machines/CI).

File: src/test/java/itm/oss/splitter/ExpenseStoreTest.java [hunk 6]
```diff
@@ -0,0 +1,170 @@
...
+
+class ExpenseStoreTest {
+
+    private ExpenseStore store;
+    private Expense sampleExpense1;
+    private Expense sampleExpense2;
+
+    // 테스트용 파일 이름을 정해둡니다. (프로젝트 폴더 최상위에 생깁니다)
+    private static final String TEST_FILE_PATH = "my_test_expenses.csv";
+
```
Comment (kwon-dohun) [inline src/test/java/itm/oss/splitter/ExpenseStoreTest.java:35]: Great setup! You could also move the repeated file deletion into this @BeforeEach only, since it appears again in @AfterEach.

File: src/test/java/itm/oss/splitter/ExpenseStoreTest.java [hunk 7]
```diff
@@ -0,0 +1,170 @@
...
+            assertFalse(scanner.hasNextLine(), "총 3줄(헤더, 데이터1, 데이터2)이어야 합니다.");
+        }
+    }
+
+    /**
+     * [테스트 3] append() - Edge Case: 참가자 목록이 비어 있는 경우
+     * -> participants 필드가 비어있는 문자열로 올바르게 변환되어야 한다.
+     */
+    @Test
+    void append_edgeCase_zeroParticipants() throws IOException {
```
Comment (kwon-dohun) [inline src/test/java/itm/oss/splitter/ExpenseStoreTest.java]: Maybe rename this test to something like append_shouldHandleEmptyParticipantsField() to describe the behavior more clearly.

**Review by kwon-dohun — COMMENTED** [review]
Good work! The tests look clean and clear.

**Comment by jobregon1212** [comment]
Hello guys, this PR should only modify the .java files (implementation and test files). It should not touch .gitignore , README, or any other file. So please fix that. Remember to synchronize with the upstream repository and pull the latest changes from the upstream repository.

File: src/test/java/itm/oss/splitter/ExpenseStoreTest.java (hunk 5 above)
Comment (ljseop1030) [reply src/test/java/itm/oss/splitter/ExpenseStoreTest.java]: Thanks for the review. Changed "try (Scanner scanner = new Scanner(testFile))" into "new Scanner(testFile, StandardCharsets.UTF_8)" for explicit charset.

File: src/main/java/itm/oss/splitter/ExpenseStore.java (hunk 1 above)
Comment (bbirribbarribbo) [reply src/main/java/itm/oss/splitter/ExpenseStore.java]: Thank you for your feedback. I've implemented your suggestion to use UTF-8 when writing the file!

File: src/main/java/itm/oss/splitter/ExpenseStore.java (hunk 3 above)
Comment (bbirribbarribbo) [reply src/main/java/itm/oss/splitter/ExpenseStore.java]: Thank you for the excellent suggestion! I changed "ArrayList" to "List" like you said. can you please take another quick look and give it your Approve??

File: src/test/java/itm/oss/splitter/ExpenseStoreTest.java (hunk 4 above)
Comment (ljseop1030) [reply src/test/java/itm/oss/splitter/ExpenseStoreTest.java:34]: Fixed the code according to your feedback. Used @TempDir to manage directories. Thanks!

File: src/test/java/itm/oss/splitter/ExpenseStoreTest.java (hunk 6 above)
Comment (ljseop1030) [reply src/test/java/itm/oss/splitter/ExpenseStoreTest.java:35]: Thanks! Deleted the redundant method under @AfterEach after adding tempDir.

File: src/main/java/itm/oss/splitter/ExpenseStore.java (hunk 2 above)
Comment (ihh25) [reply src/main/java/itm/oss/splitter/ExpenseStore.java]: Thanks for the feedback! I've added the helper method for CSV field conversion. Could you please take a look?

File: src/test/java/itm/oss/splitter/ExpenseStoreTest.java (hunk 7 above)
Comment (ihh25) [reply src/test/java/itm/oss/splitter/ExpenseStoreTest.java]: Thanks for the feedback! That's a good point. I've changed it to `append_shouldHandleEmptyParticipantsField` to make it clear it's testing the zero participants case.

**Review by kwon-dohun — APPROVED** [review]
It all looks good to me. Appreciate for applying my opinion!

File: src/test/java/itm/oss/splitter/ExpenseStoreTest.java (hunk 4 above)
Comment (ohkyounghun) [reply src/test/java/itm/oss/splitter/ExpenseStoreTest.java:34]: LGTM. Now it looks much clear and safe than before. Thank you for receiving my feedback!

**Review by ohkyounghun — APPROVED** [review]

File: src/main/java/itm/oss/splitter/ExpenseStore.java (hunk 1 above)
Comment (gunuzello) [reply src/main/java/itm/oss/splitter/ExpenseStore.java]: Great! Thanks for applying the UTF-8 update — it’ll make the file handling much safer across systems.
Everything looks good to me.

File: src/main/java/itm/oss/splitter/ExpenseStore.java (hunk 2 above)
Comment (gunuzello) [reply src/main/java/itm/oss/splitter/ExpenseStore.java]: Awesome! I checked the update — adding the CSV helper was a great call.
It makes the code cleaner and more maintainable. Looks perfect now!

File: src/main/java/itm/oss/splitter/ExpenseStore.java (hunk 3 above)
Comment (gunuzello) [reply src/main/java/itm/oss/splitter/ExpenseStore.java]: Perfect! Changing to List improves flexibility for future extensions.
The modification looks correct — I’m approving this change.

**Review by gunuzello — APPROVED** [review]
LGTM. All previous feedback has been addressed.
The code looks stable, clear, and consistent now. Approving the changes.
